File: backend/src/main/java/br/com/calciolari/datahub/sales/infrastructure/persistence/SaleItemEntity.java
package br.com.calciolari.datahub.sales.infrastructure.persistence;

import java.math.BigDecimal;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "sale_item")
public class SaleItemEntity {

	@Id
	private UUID id;

	@Column(name = "sale_id", nullable = false)
	private UUID saleId;

	@Column(name = "product_id", nullable = false)
	private UUID productId;

	@Column(name = "parse_attempt_id", nullable = false)
	private UUID parseAttemptId;

	@Column(name = "source_record_index", nullable = false)
	private int sourceRecordIndex;

	@Column(nullable = false)
	private BigDecimal quantity;

	@Column(name = "unit_price", nullable = false)
	private BigDecimal unitPrice;

	@Column(name = "discount_percentage")
	private BigDecimal discountPercentage;

	@Column(nullable = false)
	private BigDecimal total;

	@Column(name = "previous_stock")
	private BigDecimal previousStock;

	@Column(name = "resulting_stock")
	private BigDecimal resultingStock;

	protected SaleItemEntity() {
	}

	public SaleItemEntity(
			UUID id,
			UUID saleId,
			UUID productId,
			UUID parseAttemptId,
			int sourceRecordIndex,
			BigDecimal quantity,
			BigDecimal unitPrice,
			BigDecimal discountPercentage,
			BigDecimal total,
			BigDecimal previousStock,
			BigDecimal resultingStock) {
		this.id = id;
		this.saleId = saleId;
		this.productId = productId;
		this.parseAttemptId = parseAttemptId;
		this.sourceRecordIndex = sourceRecordIndex;
		this.quantity = quantity;
		this.unitPrice = unitPrice;
		this.discountPercentage = discountPercentage;
		this.total = total;
		this.previousStock = previousStock;
		this.resultingStock = resultingStock;
	}

	public UUID getId() {
		return id;
	}

	public UUID getSaleId() {
		return saleId;
	}

	public UUID getProductId() {
		return productId;
	}

	public UUID getParseAttemptId() {
		return parseAttemptId;
	}

	public int getSourceRecordIndex() {
		return sourceRecordIndex;
	}

	public BigDecimal getTotal() {
		return total;
	}
}

File: backend/src/Calciolari.DataHub/Persistence/Entities/SaleItemEntity.cs
namespace Calciolari.DataHub.Persistence.Entities;

public sealed class SaleItemEntity
{
    public Guid Id { get; set; }
    public Guid SaleId { get; set; }
    public Guid ProductId { get; set; }
    public Guid ParseAttemptId { get; set; }
    public int SourceRecordIndex { get; set; }
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal? DiscountPercentage { get; set; }
    public decimal Total { get; set; }
    public decimal? PreviousStock { get; set; }
    public decimal? ResultingStock { get; set; }

    public SaleItemEntity()
    {
    }

    public SaleItemEntity(
        Guid id,
        Guid saleId,
        Guid productId,
        Guid parseAttemptId,
        int sourceRecordIndex,
        decimal quantity,
        decimal unitPrice,
        decimal? discountPercentage,
        decimal total,
        decimal? previousStock,
        decimal? resultingStock)
    {
        Id = id;
        SaleId = saleId;
        ProductId = productId;
        ParseAttemptId = parseAttemptId;
        SourceRecordIndex = sourceRecordIndex;
        Quantity = quantity;
        UnitPrice = unitPrice;
        DiscountPercentage = discountPercentage;
        Total = total;
        PreviousStock = previousStock;
        ResultingStock = resultingStock;
    }
}
